tests/security/ima/ima_setup.pm

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Setup environmnet for IMA & EVM testing - import MOK cert,
+#          setup grub boot menu and install necessary ima tools
+# Maintainer: wnereiz <wnereiz@member.fsf.org>
+# Tags: poo#45662
+
+use base "consoletest";
+use strict;
+use testapi;
+use utils;
+use bootloader_setup 'add_grub_cmdline_settings';
+use power_action_utils "power_action";
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'root-console';
```

Comment:
I guess you prefer not using virtio console (`select_serial_terminal()`) :).
BTW I did e3fc5d807 ("security: Migrate SELinux tests to select_serial_terminal()") commit in hope that you like it and I/you would migrate rest of the security tests to virtio console. It's up to you, if you don't like it you can revert this change (probably make sense to be consistent on all security tests).

Reply:
I would like to try. I have switched `ima_setup.pm` to serial_terminal. But I can not do that for `mokutil_sign.pm`, it always failed when handling interactive commands with script_run_interactive (which I just created and submitted to lib/utils.pm), see http://10.67.17.9/tests/252#step/mokutil_sign/29

It looks like the serial console always waiting for **'#'** which is the prompt by default, but in the interactive command it is not usually use it.

Currently, I do not have enough time for more investigation. So I would like to keep this test case there. Do you have any ideas?